docs/extensibility/extending-blocks.md

```diff
@@ -38,6 +38,43 @@ wp.hooks.addFilter(
 
 A filter that applies to the result of a block's `save` function. This filter is used to replace or extend the element, for example using `wp.element.cloneElement` to modify the element's props or replace its children, or returning an entirely new element.
 
+Notably, while you _can_ modify a block's `save` result from a theme or plugin, this behavior won't persist when the theme or plugin is deactivated.
+
+_Example:_
+
+Include a custom data attribute on the Image Block `<img>` tag when the block attribute is set.
+
+```js
+function addCustomDataAttribute( element, blockType, attributes ) {
+	if ( 'core/image' !== blockType.name ) {
+		return element;
+	}
+
+	if ( ! attributes.myCustomAttribute.length ) {
+		return element;
+	}
+	// Modifying nested JavaScript objects is pretty awful.
```

Comment:
- This should probably be using `cloneElement` to override props
- Deep object assignment can be improved by using [`_.merge`](https://lodash.com/docs/4.17.10#merge) instead

Reply:
Can you provide an example?

Reply:
```js
// Bad:
return _.assign( {}, element, { myprop: 'foo' } );

// Good:
return wp.element.cloneElement( element, { myprop: 'foo' } );
```